CRM/Mailing/BAO/Mailing.php

```diff
@@ -2251,7 +2251,7 @@ public static function checkPermission($id) {
     }
 
     if (!in_array($id, $mailingIDs)) {
-      CRM_Core_Error::fatal(ts('You do not have permission to access this mailing report'));
+      CRM_Core_Error::statusBounce(ts('You do not have permission to access this mailing report'));
```

Comment:
I think I would probably throw an exception on his one on the basis that
1) the right place for a status bounce is the form layer and
2) an exception will not give a worse experience than a fatal here -  so it's not a step backwards for the user & hopefully  at some point someone will find the right place for a status bounce

Reply:
I'm ok with that and have pushed the change @eileenmcnaughton